### ISSUE TITLE
noFloatingPromise lint error

### ISSUE THREAD

**Issue by SethWen** [biomejs/biome#6985]
### Environment information

```bash
biome 2.1.2
```

### Rule name

noFloatingPromise 

### Playground link

https://biomejs.dev/playground/?lintRules=noFloatingPromises&code=ZgB1AG4AYwB0AGkAbwBuACAAZABvAFMAdABoACgAKQAgAHsACgAgACAAIAAgAGMAbwBuAHMAdAAgAGkAbgBuAGUAcgBGAG4AIAA9ACAAKAApACAAPQA%2BACAAewAKACAAIAAgACAAIAAgACAAIAByAGUAdAB1AHIAbgAgAFAAcgBvAG0AaQBzAGUALgByAGUAcwBvAGwAdgBlACgAMQApADsACgAgACAAIAAgAH0ACgAgACAAIAAgAHIAZQB0AHUAcgBuACAAJwBoAGEAaAAnAAoAfQAKAAoAZABvAFMAdABoACgAKQA%3D

### Expected result

```typescript
function doSth() {
    const innerFn = () => {
        return Promise.resolve(1);
    }
    return 'hah'
}


//   ✖ A "floating" Promise was found, meaning it is not properly handled and could lead to ignored errors or unexpected behavior.
doSth()
```

However the `doSth` is not an async function.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by dyc3** [comment]
A note to maintainers: Project rules currently don't work in the playground, so the playground won't show any diagnostics. I've tested this manually and can confirm it does throw a diagnostic.

**Comment by ematipico** [comment]
https://next.biomejs.dev/playground/?lintRules=noFloatingPromises&analyzerFixMode=safeAndUnsafeFixes&code=ZgB1AG4AYwB0AGkAbwBuACAAZABvAFMAdABoACgAKQAgAHsACgAgACAAIAAgAGMAbwBuAHMAdAAgAGkAbgBuAGUAcgBGAG4AIAA9ACAAKAApACAAPQA%2BACAAewAKACAAIAAgACAAIAAgACAAIAByAGUAdAB1AHIAbgAgAFAAcgBvAG0AaQBzAGUALgByAGUAcwBvAGwAdgBlACgAMQApADsACgAgACAAIAAgAH0ACgAgACAAIAAgAHIAZQB0AHUAcgBuACAAJwBoAGEAaAAnAAoAfQAKAAoAZABvAFMAdABoACgAKQA%3D&ruleDomains.solid=recommended
